backend/app/store/service/tree.go

```diff
@@ -106,7 +84,8 @@ func (t *Tree) proc(comments []store.Comment, node *Node, rd *recurData, parentI
 	sort.Slice(node.Replies, func(i, j int) bool {
 		return node.Replies[i].Comment.Timestamp.Before(node.Replies[j].Comment.Timestamp)
 	})
-	return node, rd.tsModified, rd.tsCreated
+	node.tsModified, node.tsCreated = rd.tsModified, rd.tsCreated
```

Comment:
This line is magic, I have no idea why it doesn't mess up tsCreated or tsModified of any of the nodes. Maybe we just don't have good enough test cases to catch it.